File: python/Progam/Layar.py
# Class Layar - Component untuk Laptop (Composition)
class Layar:
    # Konstruktor untuk inisialisasi objek
    def __init__(self, jenis="", resolusi="", touchscreen=False):
        # Atribut private (convention menggunakan underscore)
        self.__jenis = jenis              # Jenis layar (IPS, OLED, MiniLED, dll)
        self.__resolusi = resolusi        # Resolusi layar (1920x1080, dll)
        self.__touchscreen = touchscreen  # Status touchscreen (True/False)
    
    # Setter untuk mengubah nilai jenis layar
    def setJenis(self, jenis):
        self.__jenis = jenis
    
    # Setter untuk mengubah nilai resolusi
    def setResolusi(self, resolusi):
        self.__resolusi = resolusi
    
    # Setter untuk mengubah status touchscreen
    def setTouchscreen(self, touchscreen):
        self.__touchscreen = touchscreen
    
    # Getter untuk mendapatkan jenis layar
    def getJenis(self):
        return self.__jenis
    
    # Getter untuk mendapatkan resolusi layar
    def getResolusi(self):
        return self.__resolusi
    
    # Getter untuk mendapatkan status touchscreen
    def getTouchscreen(self):
        return self.__touchscreen

File: cpp/Progam/Layar.cpp
#include <bits/stdc++.h>
using namespace std;

// Class Layar - Component untuk Laptop (Composition)
class Layar{
    private:
        // Atribut private untuk encapsulation
        string jenis;          // Jenis layar (IPS, OLED, dll)
        string resolusi;       // Resolusi layar (1920x1080, dll)
        bool touchscreen;      // Status touchscreen (true/false)
    
    public:
        // Konstruktor default
        Layar(){
        }
        
        // Konstruktor dengan parameter untuk inisialisasi objek
        Layar(string jenis, string resolusi, bool touchscreen){
            this->jenis = jenis;
            this->resolusi = resolusi;
            this->touchscreen = touchscreen;
        }
        
        // Setter untuk mengubah nilai jenis layar
        void setJenis(string jenis){
            this->jenis = jenis;
        }
        
        // Setter untuk mengubah nilai resolusi
        void setResolusi(string resolusi){
            this->resolusi = resolusi;
        }
        
        // Setter untuk mengubah status touchscreen
        void setTouchscreen(bool touchscreen){
            this->touchscreen = touchscreen;
        }
        
        // Getter untuk mendapatkan jenis layar
        string getJenis(){
            return this->jenis;
        }
        
        // Getter untuk mendapatkan resolusi layar
        string getResolusi(){
            return this->resolusi;
        }
        
        // Getter untuk mendapatkan status touchscreen
        bool getTouchscreen(){
            return this->touchscreen;
        }
        
        // Destruktor untuk membersihkan memori
        ~Layar(){
        }
};